Sources/GRPC/AsyncAwaitSupport/GRPCAsyncResponseStream.swift

```diff
@@ -44,7 +44,8 @@ public struct GRPCAsyncResponseStream<Element>: AsyncSequence {
 
     @inlinable
     public mutating func next() async throws -> Element? {
-      try await self.iterator.next()
+      if Task.isCancelled { throw GRPCStatus(code: .cancelled) }
```

Comment:
Hm, I am bit unsure about that shouldn't we throw the `try checkCancellation()`  here? The `Task` got cancelled and we did not receive `GRPCStatus` from the remote and it could be confusing to the user why he got a `GRPCStatus` here.

Reply:
We do it for consistency with other throwing gRPC apis. We always produce a `GRPCStatus` _even_ if the remote did not send one (this is expected behaviour).

When cancelling client calls with `cancel()` the resulting status of the call will be `.cancelled` which is _also_ used to fail any remaining promises/continuations (i.e. any `try await call.response` will throw a `GRPCStatus(code: .cancelled)`).